pywr/recorders/recorders.py

```diff
@@ -129,7 +136,7 @@ def after(self):
         from pywr._core import Node, Storage
 
         values = [self.model.timestepper.current.datetime.isoformat()]
-        for node_name in self.node_names:
+        for node_name in self._node_names:
             node = self.model.nodes[node_name]
             if isinstance(node, Node):
```

Comment:
This should check against AbstractNode and AbstractStorage

See here for example from TablesRecorder.
https://github.com/pywr/pywr/blob/5375bb39b7cf541ee704146ffeba3ca17a6e0205/pywr/recorders/recorders.py#L458